lib/stat_tracker.rb

```diff
@@ -94,10 +94,24 @@ def average_goals_per_game #issue #8
   end
 
   def average_goals_by_season #issue #9
+    my_hash = Hash.new { |h,k| h[k.to_s] = [] }
+    count_of_games_by_season.each do |season, game_count|
+      my_hash[season] = [] 
+    end
 
-  end
-
+      count_of_games_by_season.each do |season, game_count|
+        game_sum_calc = []
+        games.each do |row| 
+          game_sum_calc << (row[:away_goals] + row[:home_goals]) if row[:season] == season
+          my_hash[season.to_s] = (game_sum_calc.sum / game_count.to_f).round(2)
+        end
+      end
 
+      my_hash.each do |key, value| 
+        my_hash.delete(key) if value == [] 
```

Comment:
Can you explain this part to me? I'm confused about why we are deleting things we just added in the code above.